appinventor/buildserver/src/com/google/appinventor/buildserver/DexExecTask.java

```diff
@@ -100,42 +103,40 @@ private boolean preDexLibraries(List<File> inputs) {
             return true;
         }
 
-        synchronized(semaphore) {
+        synchronized (semaphore) {
 
             final int count = inputs.size();
-            boolean allSuccessful = true;
-            for (int i = 0 ; i < count; i++) {
+            boolean successful = true;
+            for (int i = 0; i < count && successful; i++) {
                 File input = inputs.get(i);
                 if (input.isFile()) {
                     // check if this libs needs to be pre-dexed
                     String fileName = getDexFileName(input);
                     File dexedLib = new File(mDexedLibs, fileName);
                     String dexedLibPath = dexedLib.getAbsolutePath();
 
-                    if (dexedLib.isFile() == false/*||
+                    if (!dexedLib.isFile()/*||
                                                     dexedLib.lastModified() < input.lastModified()*/) {
 
                         System.out.println(
-                            String.format("Pre-Dexing %1$s -> %2$s",
-                              input.getAbsolutePath(), fileName));
+                                String.format("Pre-Dexing %1$s -> %2$s",
+                                        input.getAbsolutePath(), fileName));
 
                         if (dexedLib.isFile()) {
                             dexedLib.delete();
                         }
-
-                        boolean dexSuccess = runDx(input, dexedLibPath, false /*showInput*/);
-                        allSuccessful = allSuccessful && dexSuccess;
+                        successful = runDx(input, dexedLibPath, /*showInputs=*/ false);
```

Comment:
Since this is going to escape out of the loop on the next iteration anyway, and since the dexing failed (so line 136 shouldn't happen), would it not make sense to return false here?

Reply:
Yeah, that's reasonable to return false there. I have made the necessary changes